core/internal/features_test.go

```diff
@@ -317,12 +317,12 @@ func TestIntegration_ExternalAdapter_RunLogInitiated(t *testing.T) {
 	gethClient.On("ChainID", mock.Anything).Return(app.Config.ChainID(), nil)
 	sub.On("Err").Return(nil).Maybe()
 	sub.On("Unsubscribe").Return(nil).Maybe()
-	newHeads := make(chan<- *models.Head, 10)
+	newHeadsCh := make(chan chan<- *models.Head, 10)
 	logsCh := cltest.MockSubscribeToLogsCh(gethClient, sub)
 	rpcClient.On("CallContext", mock.Anything, mock.Anything, mock.Anything, mock.Anything, mock.Anything).Return(nil)
 	rpcClient.On("EthSubscribe", mock.Anything, mock.Anything, "newHeads").
 		Run(func(args mock.Arguments) {
-			newHeads = args.Get(1).(chan<- *models.Head)
+			newHeadsCh <- args.Get(1).(chan<- *models.Head)
```

Comment:
@RyanRHall I think this is the cause of the flakiness, the Run wasn't blocking

Reply:
:+1: :+1: :+1: